runtime/services/activation/activation.cc

```diff
@@ -1,111 +1,121 @@
+  if (is_open) {
+    int id = rand() % files_from_flora.size();
+    prepareWavPlayer(files_from_flora[id].c_str(), "system", true);
+    if (!volume_set) {
       char val[PROP_VALUE_MAX];
-      property_get("persist.audio.volume.system", (char*)&val, "");
+      property_get("persist.audio.volume.system", (char *) &val, "");
       int vol = atoi(val);
       fprintf(stdout, "init activation volume to %d\n", vol);
       rk_set_stream_volume(STREAM_SYSTEM, vol);
       volume_set = true;
     }
-
-    startWavPlayer();
-  }
-  // cppcheck-suppress unusedFunction
-  void disconnected() {
-    thread tmp([this]() { this->flora_disconnected(); });
-    tmp.detach();
   }
-  void flora_disconnected() {
-    flora_cli.reset();
-    reconn_mutex.lock();
-    reconn_cond.notify_one();
-    reconn_mutex.unlock();
-  }
-  void start() {
-    shared_ptr<Client> cli;
-    unique_lock<mutex> locker(reconn_mutex);
+}
 
-    while (true) {
-      int32_t r = Client::connect("unix:/var/run/flora.sock", this, 0, cli);
-      if (r != FLORA_CLI_SUCCESS) {
-        fprintf(stderr, "init flora client failed, please retry\n");
-        reconn_cond.wait_for(locker, chrono::seconds(5));
-      } else {
-        cli->subscribe("rokid.turen.voice_coming");
-        flora_cli = cli;
-        reconn_cond.wait(locker);
-      }
+void Activation::playAwake() {
+  char propValue[PROP_VALUE_MAX];
+  property_get("persist.dndmode.awakeswitch", (char *) propValue, "");
+  if (strcmp(propValue, "open") != 0) {
+    fprintf(stdout, "dnd mode, just skip\n");
+    return;
+  } else {
+    property_get("state.network.connected", (char *) propValue, "");
+    if (strcmp(propValue, "true") != 0) {
+      fprintf(stdout, "current network is not available, just skip\n");
+      return;
     }
   }
+  if (is_open) {
+    startWavPlayer();
+    prepareForNextAwake();
+  }
+}
 
- private:
-  bool volume_set = false;
-  shared_ptr<Client> flora_cli;
-  mutex reconn_mutex;
-  condition_variable reconn_cond;
-};
-
-int main(int argc, char** argv) {
-  srand(time(NULL));
-  prePrepareWavPlayer(filenames, 4);
-  fprintf(stdout, "wav player has been preloaded all activation files\n");
+#define CAPS_READ(action) if (action != CAPS_SUCCESS) goto ERROR
 
-  Activation activation;
-  activation.start();
-  return 1;
+void Activation::applyAwakeSound(shared_ptr <Caps> &msg) {
+  int32_t fCount;
+  CAPS_READ(msg->read(fCount));
+  files_from_flora.clear();
+  if (fCount > MAX_PLAY_LIST)
+    fCount = MAX_PLAY_LIST;
+  const char *filename_list[MAX_PLAY_LIST];
+  for(int i = 0; i < fCount; ++i) {
+    files_from_flora.emplace_back();
+    CAPS_READ(msg->read_string(files_from_flora.back()));
+    filename_list[i] = files_from_flora[i].c_str();
+  }
+  is_open = fCount > 0;
+  if (is_open) {
+    prePrepareWavPlayer(filename_list, files_from_flora.size());
+    fprintf(stdout, "wav player has been preloaded all activation files\n");
+    prepareForNextAwake();
+  }
+  return;
+ERROR:
```

Comment:
Unused label.

Reply:
ERROR was used in macro which is defined above